src/cubic-b-spline.cc

```diff
@@ -231,6 +231,10 @@ namespace trajectory
 
 	while (!found)
 	  {
+	   if (t == knots_[static_cast<size_t>(imax+1)]) //We are exactly at the end of the spline
+	   {
+	     return imax;
+	   }
```

Comment:
If the problem only concerns `t = t_max`, this check should not be in the inner `while` loop either, since this will be checked at every iteration of the dichotomy.